### PR TITLE
[3.0] add constexpr support to varint types

### PR DESCRIPTION
**Pull request by linh2931**
## Change Description

Resolve https://github.com/AntelopeIO/cdt/issues/13

Based on https://github.com/AntelopeIO/cdt/pull/14. Thanks @smlu. To conform to our practice, fix release branch and then merge the fix to main.

Add comprehensive unit test cases.

## API Changes
- [ ] API Changes

## Documentation Additions
- [ ] Documentation Additions

### CODE REVIEW

File: tests/unit/varint_tests.cpp [hunk 1]
```diff
@@ -252,6 +252,194 @@ EOSIO_TEST_BEGIN(signed_int_type_test)
...
+
+   // ----------------------------------------------------------
+   // friend bool operator>=(const uint32_t&, const signed_int&)
+   static_assert( int32_t{42} >= signed_int{42} );
+   static_assert( (int32_t{42} >= signed_int{43}) == false );
+
+   // ------------------------------------------------------------
+   // friend bool operator>=(const signed_int&, const signed_int&)
+   static_assert( signed_int{42} >= signed_int{42} );
+   static_assert( (signed_int{42} >= signed_int{43}) == false );
```
Comment (smlu) [inline tests/unit/varint_tests.cpp:440]: Nit, I'd add more 2,3 & 4 bytes int/uint tests with casts or integer literals i.e. `xU`, `xL` etc.. and `std::size_t`, just to make sure everything passes at compile time (which of course should).

Comment (linh2931) [reply tests/unit/varint_tests.cpp]: Thanks. More tests were added.

File: libraries/eosiolib/core/eosio/varint.hpp [hunk 2]
```diff
@@ -76,7 +76,7 @@ namespace eosio {
         * @return true - if equal
         * @return false - otherwise
         */
-       friend bool operator==( const unsigned_int& i, const uint32_t& v )     { return i.value == v; }
+       constexpr friend bool operator==( const unsigned_int& i, const uint32_t& v )     { return i.value == v; }
```
[the remark this answers is not in the document]
Comment (linh2931) [reply libraries/eosiolib/core/eosio/varint.hpp:79]: The whitespaces had been there before. That's why I did not touch them.

[the remark this answers is not in the document]
Comment (linh2931) [reply libraries/eosiolib/core/eosio/varint.hpp:79]: I decided to take the whitespaces out. Thanks.

**Review by larryk85 — APPROVED** [review]

**Pull request merged**